core/connection.py

```
@@ -47,8 +47,24 @@ def _path(self, client_id: str) -> str:
+            try:
+                with open(tmp, "w") as f:
+                    json.dump(list(details), f)
+                    f.flush()
+                    os.fsync(f.fileno())
+                os.replace(tmp, path)
+            except Exception:
+                # Don't leave an orphaned temp file behind on failure.
+                try:
+                    os.remove(tmp)
+                except OSError:
+                    pass
```

Ruff (0.15.21)

[warning] Use `contextlib.suppress(OSError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(OSError): ...`

(SIM105)